store/src/java/com/zimbra/cs/service/util/VolumeConfigUtil.java

```diff
@@ -152,35 +171,62 @@ public static void validateCreateVolumeRequest(CreateVolumeRequest request, Volu
             } else if (AdminConstants.A_VOLUME_OPEN_IO.equalsIgnoreCase(storageTypeOpenIO)) {
                 // validate proxy port as positive
                 if (volInfoRequest.getVolumeExternalOpenIOInfo().getProxyPort() <= 0) {
-                    throw VolumeServiceException.INVALID_REQUEST("Proxy port can't be negative number or null", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS_THRESHOLD);
+                    throw VolumeServiceException.INVALID_REQUEST("Proxy port can't be negative number or 0", VolumeServiceException.BAD_VOLUME_PROXY_PORT);
                 }
 
                 // validate account port as positive
                 if (volInfoRequest.getVolumeExternalOpenIOInfo().getAccountPort() <= 0) {
-                    throw VolumeServiceException.INVALID_REQUEST("Account port can't be negative number or null", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS_THRESHOLD);
+                    throw VolumeServiceException.INVALID_REQUEST("Account port can't be negative number or 0", VolumeServiceException.BAD_VOLUME_ACCOUNT_PORT);
                 }
 
                 // validate url is empty or not
                 if (StringUtil.isNullOrEmpty(volInfoRequest.getVolumeExternalOpenIOInfo().getUrl())) {
-                    throw VolumeServiceException.INVALID_REQUEST("Url can't be null", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS_THRESHOLD);
+                    throw VolumeServiceException.INVALID_REQUEST("Url can't be null", VolumeServiceException.BAD_VOLUME_URL);
+                }
+
+                // validate format of url
+                if (!volInfoRequest.getVolumeExternalOpenIOInfo().getUrl().contains("http:/") && !volInfoRequest.getVolumeExternalOpenIOInfo().getUrl().contains("https:/")) {
```

Comment:
are we allowing urls like below
`https://some weird url.com/`

Reply:
here we are checking only format

Reply:
that I can see what we are checking, but my question is different

Reply:
https://github.com/Zimbra/zm-hsm-store/blob/develop/src/java/com/zimbra/cs/admin/util/S3BucketGlobalConfigUtil.java#L47

Reply:
Thanks Rakesh .. updated in the code for url verification